src/server/api/routers/smart-home/entity-state.ts

```diff
@@ -0,0 +1,53 @@
+import { TRPCError } from '@trpc/server';
+
+import { ZodError, z } from 'zod';
+
+
+import { createTRPCRouter, protectedProcedure } from '../../trpc';
+
+import { findAppProperty } from '~/tools/client/app-properties';
+import { getConfig } from '~/tools/config/getConfig';
+import { HomeAssistantSingleton } from '~/tools/singleton/HomeAssistantSingleton';
+
+export const smartHomeEntityStateRouter = createTRPCRouter({
+  retrieveStatus: protectedProcedure
+    .input(
+      z.object({
+        configName: z.string(),
+        entityId: z.string().regex(/^[A-z0-9-_\.]+$/)
+      })
+    )
+    .query(async ({ input }) => {
+      const config = getConfig(input.configName);
+
+      const instances = config.apps.filter((app) => app.integration?.type == 'homeAssistant');
+
+      for (var instance of instances) {
+        const url = new URL(instance.url);
+        const client = HomeAssistantSingleton.getOrSet(url, findAppProperty(instance, 'apiKey'));
+        const state = await client.getEntityState(input.entityId);
+        
+        if (!state.success) {
+          if (!(state.error instanceof ZodError)) {
+            continue;
+          }
+          // Consola.error('Unable to handle entity state: ', state.error);
+          throw new TRPCError({
+            code: 'NOT_IMPLEMENTED',
+            message: `Unable to handle entity state. Check log for details`
+          });
+        }
+
+        if(!state.data) {
+          throw new TRPCError({
+            code: 'INTERNAL_SERVER_ERROR',
+            message: `Unable to connect to app '${instance.id}'. Check logs for details`
```

Comment:
App is not descriptive, the person won’t understand the error or what widget is causing it

Reply:
I am currently unable to display more information. I think adding more verbose information will not be nessesary anymore once we have the centralized integration panel. I'd recommend that we merge it like this.

Reply:
You’re not using trpc correctly, having an integration panel won’t change anything. Unable to connect to app 278UebUwnei728hnwi% will not help anyone that is encountering an issue with this widget

Reply:
> You’re not using trpc correctly, having an integration panel won’t change anything. Unable to connect to app 278UebUwnei728hnwi% will not help anyone that is encountering an issue with this widget

We already log implicitly in the SDK. Why log twice?

Reply:
I mean the error message you’re throwing doesn’t really help, it could at least say what widget it’s coming from, people will send us logs with this error and we won’t know where it’s from

Reply:
I think we should make a WidgetError type that wraps the errors nicely